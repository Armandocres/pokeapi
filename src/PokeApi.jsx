import React, { useState } from "react";
import AddPokemon from "./components/AddPokemon";
import PokeGrid from "./components/PokeGrid";

const PokeApi = ({ defaultCategpries = "pikachu" }) => {
	const [pokemon, setPokemon] = useState(defaultCategpries);

	return (
		<>
			<h1>PokeApi</h1>
			<AddPokemon setpoke={setPokemon} />
			<PokeGrid pokemon={pokemon} />
		</>
	);
};

export default PokeApi;
