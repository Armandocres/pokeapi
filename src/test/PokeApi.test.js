import React from 'react';
import PokeApi from '../PokeApi';
import { shallow } from 'enzyme';

describe('Pruebas en el componentes giftExpertApp', () => {

  test('Prueba del snapchot', () => {
    const wrapper = shallow(<PokeApi />);
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe mostrar la lista de categorias', () => {
    const categories = 'Dragon Ball';
    const warapper = shallow(<PokeApi defaultCategories={categories} />);

    expect(warapper).toMatchSnapshot();
  })

})