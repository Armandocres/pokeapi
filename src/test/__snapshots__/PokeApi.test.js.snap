// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componentes giftExpertApp Debe mostrar la lista de categorias 1`] = `
<Fragment>
  <h1>
    PokeApi
  </h1>
  <AddPokemon
    setpoke={[Function]}
  />
  <PokeGrid
    pokemon="pikachu"
  />
</Fragment>
`;

exports[`Pruebas en el componentes giftExpertApp Prueba del snapchot 1`] = `
<Fragment>
  <h1>
    PokeApi
  </h1>
  <AddPokemon
    setpoke={[Function]}
  />
  <PokeGrid
    pokemon="pikachu"
  />
</Fragment>
`;
